src/cloudConsole/cloudConsole.ts

```diff
@@ -343,36 +343,54 @@ export function createCloudConsole(subscriptionProvider: AzureSubscriptionProvid
             liveServerQueue = serverQueue;
 
             const tenants = await subscriptionProvider.getTenants();
-            serverQueue.push({ type: 'log', args: [localize('foundTenants', `Found ${tenants.length} tenant${tenants.length > 1 ? 's' : ''}.`)] });
             let selectedTenant: TenantIdDescription | undefined = undefined;
-            // handle multi tenant scenario, user must pick a tenant
-            if (tenants.length > 1) {
-                serverQueue.push({ type: 'log', args: [localize('selectingTenant', `Selecting tenant...`)] });
-                const picks = tenants.map(tenant => {
-                    const defaultDomainName: string | undefined = tenant.defaultDomain;
-                    return <IAzureQuickPickItem<TenantIdDescription>>{
-                        label: tenant.displayName,
-                        description: defaultDomainName,
-                        data: tenant,
-                    };
-                }).sort((a, b) => a.label.localeCompare(b.label));
-
-                const pick = await window.showQuickPick<IAzureQuickPickItem<TenantIdDescription>>(picks, {
-                    placeHolder: localize('selectDirectoryPlaceholder', "Select tenant"),
-                    ignoreFocusOut: true // The terminal opens concurrently and can steal focus (https://github.com/microsoft/vscode-azure-account/issues/77).
+
+            if (tenants.length <= 1) {
```

Comment:
Is there any reason to have this conditional? It seems like you basically end up doing the same thing in `if (tenantsWithSubs.length <= 1)`

Reply:
This is step 1 in the steps I outlined in the description.

I think it's very common to only have one tenant. Knowing that, I'd like to avoid making extra ARM calls wherever possible.

I could remove this shortcut case to simplify the code, but I think it's worth keeping since it speeds up the operation and avoids an ARM call for the common case.

Reply:
> I think it's very common to only have one tenant. Knowing that, I'd like to avoid making extra ARM calls wherever possible.

Yeah, I figured that one ARM call for one tenant wouldn't be that bad to make the code simpler, but your reasoning makes sense to me too.